CONTRIBUTING.md

```diff
@@ -55,32 +29,27 @@ If you don't know what contribution you can work on, here are a few suggestions:
 * Write documentation or fix the existing documentation
 * If you still don't know what could be a good task for you, do not hesitate to contact us.
 
+# We will close your PR if
 
-## Testing
-
-For our TypeScript codebase, you can run the entire test suite using `yarn test` in the root directory.
-
-For our Rust codebase, you can run the test suites for each project by running `cargo test` in the project directory.
-
-## Continuous integration
-
-After creating a PR on GitHub, the code will be tested automatically by GitHub Action. The tests can take up to 15 minutes to pass. We ask you to test your code on your machine before submitting a PR.
-
-
-## Style Guide
-
-Iron Fish uses `eslint` and `prettier` to maintain consistent formatting on the TypeScript codebase.
-For the Rust codebase, we are using `rustfmt`.
+If you are working on something in one of these categories, we will not accept your PR if you don't open a Fish Improvement Proposal (FIP).
 
-Please run it before submitting a change.
+ - Upgrading Package Versions
+   - This is a common attack vector by malicious agents. We lock down our packages for this very reason, and only allolw the core team to upgrade packages unlesse you submit a FIP first explaining why you want to upgrade the package.
+ - Upgrading Node Versions
+   - There is a large impact  in upgrading node versions. Iron Fish takes advantage of many experimental node features that at some are not fully fleshedout. We use workers, and native code boundries. Node is also known to introduce bugs in newer versions, even LTS. Because of this, there is production testing we do on our side when upgrading node to ensure it's compatible. Because of this, we don't allow usersc to upgrade our node versions unless a FIP is filed.
```

Comment:
Nit: Extra space after "large impact"; "fleshedout" -> "fleshed out", "boundries" -> "boundaries", "usersc" -> "users"

Also, consider "...experimental node features, some of which are not fully fleshed out."